Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -160,6 +161,169 @@ public void DispatchValidationFailures_WithRowEditModeControlHavingInvalidValida
+      Assert.That(failuresInRepository.Single().Failure.ErrorMessage, Is.EqualTo("The value contains unsupported characters: \"\u0001\" (position: 1)."));
+
+      validationResultMock.Verify();
+
+      // Validates that dispatch was called to dispatching validator
+      var bocTextValue = ((Control)editableRow).FindControl("0") as BusinessObjectBoundEditableWebControl;
+      validationResultMock.Verify(_ => _.GetValidationFailures(bocTextValue.DataSource.BusinessObject, bocTextValue.Property, true));
+    }
+
+    [Test]
+    public void DispatchValidationFailures_WithoutErrorIndicatorColumn_DoesNotDispatchToRepository ()
+    {
+      IBusinessObject businessObject = TypeWithReference.Create();
+      var businessObjectProperty = businessObject.BusinessObjectClass.GetPropertyDefinition("ReferenceList");
+
+      var bocList = new BocListMock();
+      bocList.ID = "someBocList";
+      bocList.DataSource = new StubDataSource(businessObject.BusinessObjectClass);
+      bocList.DataSource.BusinessObject = businessObject;
+      bocList.Property = (IBusinessObjectReferenceProperty)businessObjectProperty;
+
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>(MockBehavior.Strict);
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ControlToValidate = "someBocList";
+
+      NamingContainer.Controls.Add(bocList);
+      NamingContainer.Controls.Add(validator);
+
+      validator.DispatchValidationFailures(validationResultMock.Object);
+
+      Assert.That(bocList.ValidationFailureRepository.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true), Is.Empty);
+    }
+
+    [Test]
+    public void DispatchValidationFailures_WithErrorIndicatorColumn_DispatchesToRepository ()
+    {
+      var validationFailureRepo = new BocListValidationFailureRepository();
+
+      var error = "BocList has validation errors.";
+      var resourceManagerStub = new Mock<IResourceManager>();
+      resourceManagerStub.Setup(_ => _.TryGetString("Remotion.ObjectBinding.Web.UI.Controls.BocList.ValidationFailuresFoundInListErrorMessage", out error)).Returns(true);
+
+      IBusinessObject businessObject = TypeWithReference.Create();
+      var businessObjectProperty = businessObject.BusinessObjectClass.GetPropertyDefinition("ReferenceList");
+
+      var editModeControllerStub = new Mock<IEditModeController>();
+      var dataSourceStub = new Mock<IBusinessObjectDataSource>();
+      dataSourceStub.Setup(_ => _.BusinessObject).Returns(businessObject);
+
+      var columnDefinitionMock = new Mock<BocColumnDefinition>();
+      var validationFailureMatcherMock = new Mock<IValidationFailureMatcher>();
+      columnDefinitionMock.As<IBocColumnDefinitionWithValidationSupport>().Setup(_ => _.GetValidationFailureMatcher()).Returns(validationFailureMatcherMock.Object);
+
+      var bocListMock = new Mock<Control> { CallBase = true };
```

Comment:
Can we simplify this setup here by using `BocListStubMock` instead? Haven't thought it through but it might do a lot of things for us that we now have an explicit setup for.

Reply:
Im not sure what a BocListStubMock is, but taking a guess that you mean the BocListMock class: That does not work. As I did this some time ago I'm not quite sure why exactly, but there were problems with setting some specific things on that object.